src/util/amount.rs

```diff
@@ -68,22 +76,33 @@ impl FromStr for Denomination {
 
     /// Convert from a str to Denomination.
     ///
-    /// Any combination of upper and/or lower case, excluding uppercase 'M' is considered valid.
-    /// - Singular: BTC, mBTC, uBTC
+    /// Any combination of upper and/or lower case, excluding uppercase of SI(m, u, n, p) is considered valid.
+    /// - Singular: BTC, mBTC, uBTC, nBTC, pBTC
     /// - Plural or singular: sat, satoshi, bit, msat
     ///
-    /// Due to ambiguity between mega and milli we prohibit usage of leading capital 'M'.
+    /// Due to ambiguity between mega and milli, pico and peta we prohibit usage of leading capital 'M', 'P'.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         use self::ParseAmountError::*;
+        use self::Denomination as D;
+
+        denomination_from_str(s).map_or_else(
+            || Err(UnknownDenomination(s.to_owned())),
+            |d| {
+                if s.starts_with(|ch : char| ch.is_uppercase() ) {
+                    match d {
+                        D::MilliBitcoin | D::PicoBitcoin | D::MilliSatoshi =>
+                            Err(PossiblyConfusingDenomination(s.to_owned())),
+                        D::NanoBitcoin | D::MicroBitcoin =>
+                            Err(UnknownDenomination(s.to_owned())),
```

Comment:
Its a little odd (to me at least) to see `=>` without a `{`. I see there are other instances of that in our code base, I'm going to patch them :) Perhaps this is cleaner if written as:
```
        denomination_from_str(s).map_or_else(
            || Err(UnknownDenomination(s.to_owned())),
            |d| {
                if s.starts_with(|ch : char| ch.is_uppercase() ) {
                    match d {
                        D::MilliBitcoin | D::PicoBitcoin | D::MilliSatoshi => {
                            Err(PossiblyConfusingDenomination(s.to_owned()))
                        },
                        D::NanoBitcoin | D::MicroBitcoin => {
                            Err(UnknownDenomination(s.to_owned()))
                        },
                        _ => Ok(d)
                    }
                } else {
                    Ok(d)
                }
            }
        )
```

Reply:
hmm, I did the patch to remove all these codebase wide but it feels a bit too much like code churn. Feel free to ignore my suggestion :)

Reply:
Thank you for the detail suggestion. I added the test case u mentioned above. But for this formatting, I actually ran `cargo fmt` and it returns this

```
                    match d {
                        D::MilliBitcoin | D::PicoBitcoin | D::MilliSatoshi => {
                            Err(PossiblyConfusingDenomination(s.to_owned()))
                        }
                        D::NanoBitcoin | D::MicroBitcoin => Err(UnknownDenomination(s.to_owned())),
                        _ => Ok(d),
                    }
```

So it looks like suggesting having not bracket when it can be fit to one line and vice versa.(btw is there any reason we are using cargo fmt?).

Reply:
Yes, years of using rustfmt is why this jumped out at me. We don't use rustfmt in this project because the project predates rustfmt becoming popular/stable. Some folks want to introduce it but its not trivial to do so.